src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/GeneratedCodeRecognition/AbstractGeneratedCodeRecognitionService.cs

```diff
@@ -11,10 +12,22 @@ namespace Microsoft.CodeAnalysis.GeneratedCodeRecognition
 {
     internal abstract class AbstractGeneratedCodeRecognitionService : IGeneratedCodeRecognitionService
     {
+#if !CODE_STYLE
         public bool IsGeneratedCode(Document document, CancellationToken cancellationToken)
         {
             var syntaxTree = document.GetSyntaxTreeSynchronously(cancellationToken);
```

Comment:
`GetSyntaxTreeSynchronously` is internal API on Document that is not available in shared layer. Shared layer should use `IsGeneratedCodeAsync` async API.

Reply:
can we file an issue to move everything to IsGeneratedCodeAsync ?  it's not clear to me why we have/need a sync path for this.

Reply:
Sure, tracked with another follow up item in https://github.com/dotnet/roslyn/issues/41462